### PR TITLE
cmd/Bosun: enable group by interval in the influx() query

### PR DESCRIPTION
**Pull request by giganteous**
A first throw at fixing the inconsistencies. See bug #1298. I think I've fixed the tests. Fighting with travis now.

### CODE REVIEW

**Comment by gbrayut** [comment]
If you rebase this on origin/master travis CI should start working. There is a new version of Esc as well so you will need to run `go get -u github.com/mjibson/esc` if you need to make any changes to the static content.

File: cmd/bosun/expr/influx.go [hunk 1]
```diff
@@ -25,16 +25,27 @@ var Influx = map[string]parse.Func{
...
+		return nil, err
+	}
+	s, ok := st.(*influxql.SelectStatement)
+	if !ok {
+		return nil, fmt.Errorf("influx: expected select statement")
+	}
+
+	t := make(parse.Tags, len(s.Dimensions))
+	for _, d := range s.Dimensions {
+		if _, ok := d.Expr.(*influxql.Call); ok { // && call.Name == "time" {
```
Comment (gbrayut) [inline cmd/bosun/expr/influx.go]: Remove the commented out code // && call.Name == "time" {

**Comment by gbrayut** [comment]
I left a note on a line that has some commented out code that should be removed.

Also the docs from https://github.com/bosun-monitor/bosun/issues/1298#issuecomment-139345964 should be commited into the /docs/expressions.md file as part of this PR, as we usually like to have doc changes in the same PR that the code is changed.

Other than that I tested this on my local instance and it looks good. The following query based on a counter matches what I would expect to get from OpenTSDB:
```
influx("opentsdb", '''SELECT non_negative_derivative(value) FROM "os.cpu" WHERE host =~ /gb.*/ GROUP BY host''', "30m", "", "1s")
```

Still not sure exactly how downsampling or fill values work for counters in Influx. If I change 1s to something larger the graph jumps above 100%, so I assume I would need to wrap the above query in another query if I wanted to get the equivalent of a 5m-avg downsample. I looks like Influx does this using something called [Continuous Queries](https://influxdb.com/docs/v0.9/query_language/continuous_queries.html).

**Comment by gbrayut** [comment]
Found a github issue about [using rates in influxdb](https://github.com/influxdb/influxdb/issues/3976#issuecomment-138755447). looks like the non_negative_derivative function can take an optional argument and aggregate function. So on my test system that sends scollector data to both influxdb and opentsdb the following queries create the same graphs:

```
influx("opentsdb", '''SELECT non_negative_derivative(mean(value),1s) FROM "os.cpu" GROUP BY host''', "30m", "", "2m")

q("sum:2m-avg:rate{counter,,1}:os.cpu{host=*}", "30m", "")
```

**Comment by giganteous** [comment]
Thanks! I'll rebase the work, and fix the issues.

**Comment by gbrayut** [comment]
The docs look good here. Could probably use another rebase to pull in the new influx settings in the conf file, but otherwise LGTM.

**Comment by giganteous** [comment]
Rebased.

I'm running our proof of concept system with the rebased code, and our version of the os.diskspace check looks to be okay.

**Comment by gbrayut** [comment]
LGTM

Thanks for working on this!

**Pull request merged**